### PR TITLE
Feature/services

### PR DESCRIPTION
**Pull request by exandron**
fix controllers pages, pom.xml, Optional

### CODE REVIEW

File: practise/pom.xml [hunk 1]
```diff
@@ -120,6 +129,32 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven.compiler.plugin.version}</version>
+                <configuration>
+                    <source>21</source>
```
Comment (egorzhurov) [inline practise/pom.xml]: ${java.version}

File: practise/src/main/java/com/modsen/practise/dto/ResponseProductDTO.java [hunk 2]
```diff
@@ -26,8 +27,6 @@ public class ProductDTO {
 
     @NotNull(message = "Category is mandatory")
```
Comment (egorzhurov) [inline practise/src/main/java/com/modsen/practise/dto/ResponseProductDTO.java]: ?????

File: practise/src/main/java/com/modsen/practise/dto/RequestOrderDTO.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.modsen.practise.dto;
+
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import lombok.*;
+
+import java.util.List;
+
+@Getter
```
Comment (egorzhurov) [inline practise/src/main/java/com/modsen/practise/dto/RequestOrderDTO.java]: Data

File: practise/src/main/java/com/modsen/practise/repository/CategoryRepository.java [hunk 4]
```diff
@@ -4,6 +4,11 @@
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+
 @Repository
 public interface CategoryRepository extends JpaRepository<Category, Long> {
+    Category findByName(String name);
```
Comment (egorzhurov) [inline practise/src/main/java/com/modsen/practise/repository/CategoryRepository.java]: Optional<Category>

File: practise/src/main/java/com/modsen/practise/repository/OrderRepository.java [hunk 5]
```diff
@@ -2,11 +2,15 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
+import java.util.Optional;
 
 @Repository
 public interface OrderRepository extends JpaRepository<Order, Long> {
+    @Query("SELECT o FROM Order o")
```
Comment (egorzhurov) [inline practise/src/main/java/com/modsen/practise/repository/OrderRepository.java]: Remove it

**Comment by norog6** [comment]
add services

**Pull request merged**